src/tools/extract-conditionals.cpp

```diff
@@ -0,0 +1,991 @@
+                                             osmium::object_id_type to,
+                                             const std::string &tag,
+                                             int value,
+                                             const std::string &condition) {
+        stream << from << "," << to << "," << tag << "," << value << ",\"" << condition << "\""
+               << "," << std::setprecision(6) << location.lon() << "," << std::setprecision(6)
+               << location.lat() << "\n";
+    });
+
+    return EXIT_SUCCESS;
+}
+
+// Time zone shape polygons loaded in R-tree
+// local_time_t is a pair of a time zone shape polygon and the corresponding local time
+// rtree_t is a lookup R-tree that maps a geographic point to an index in a local_time_t vector
+using point_t = boost::geometry::model::
+    point<double, 2, boost::geometry::cs::spherical_equatorial<boost::geometry::degree>>;
+using polygon_t = boost::geometry::model::polygon<point_t>;
+using box_t = boost::geometry::model::box<point_t>;
+using rtree_t =
+    boost::geometry::index::rtree<std::pair<box_t, size_t>, boost::geometry::index::rstar<8>>;
+using local_time_t = std::pair<polygon_t, struct tm>;
+
+// Function loads time zone shape polygons, computes a zone local time for utc_time,
+// creates a lookup R-tree and returns a lambda function that maps a point
+// to the corresponding local time
+auto LoadLocalTimesRTree(const std::string &tz_shapes_filename, std::time_t utc_time)
+{
+    // Load time zones shapes and collect local times of utc_time
+    auto shphandle = SHPOpen(tz_shapes_filename.c_str(), "rb");
+    auto dbfhandle = DBFOpen(tz_shapes_filename.c_str(), "rb");
+
+    if (!shphandle || !dbfhandle)
+    {
+        throw osrm::util::exception("failed to open " + tz_shapes_filename + ".shp or " +
+                                    tz_shapes_filename + ".dbf file");
+    }
+
+    int num_entities, shape_type;
+    SHPGetInfo(shphandle, &num_entities, &shape_type, NULL, NULL);
+    if (num_entities != DBFGetRecordCount(dbfhandle))
+    {
+        throw osrm::util::exception("inconsistent " + tz_shapes_filename + ".shp and " +
+                                    tz_shapes_filename + ".dbf files");
+    }
+
+    const auto tzid = DBFGetFieldIndex(dbfhandle, "TZID");
+    if (tzid == -1)
+    {
+        throw osrm::util::exception("did not find field called 'TZID' in the " +
+                                    tz_shapes_filename + ".dbf file");
+    }
+
+    // Lambda function that returns local time in the tzname time zone
+    // Thread safety: MT-Unsafe const:env
+    std::unordered_map<std::string, struct tm> local_time_memo;
+    auto get_local_time_in_tz = [utc_time, &local_time_memo](const char *tzname) {
+        auto it = local_time_memo.find(tzname);
+        if (it == local_time_memo.end())
+        {
+            struct tm timeinfo;
+            setenv("TZ", tzname, 1);
```

Comment:
This comes from POSIX interface  http://man7.org/linux/man-pages/man3/tzset.3.html "The tzset() function initializes the tzname variable from the TZ environment variable." 

Will look how to safely change `tzname` without TZ envar modification.

Hard solution: parse timezone /usr/share/zoneinfo/ but it will be just non-portable reimplementation